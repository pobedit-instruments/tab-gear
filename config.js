let date = new Date();

export default {
	email: 'podebit.ms@gmail.com',
	legal: 'Pobedit Instuments',
	year : date.getFullYear()
};
