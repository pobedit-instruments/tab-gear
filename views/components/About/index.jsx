import React, { Component, PropTypes } from 'react';

import Link from '../Link';
import Text from '../Text';
// import Share from '../Share';

class About extends Component {
	constructor (properties) {
		super(...arguments);
	}

	componentDidMount () {
		window.scrollTo(0, 0);
	}

	getFeatures () {
		let features = [
			'tab_list',
			'search',
			'incognito',
			'discarding',
			'moving',
			'close_tabs',
			'close_all_tabs',
			'total_tabs',
			'open_extensions',
			'i18n',
			'shortcuts',
			'prevent_closing',
			'highlighting',
			// 'confirmation'
			// 'navigation',
		];

		return features.map((feature, index) => {
			return <li className="tg-list__item" key={ index }>
						{ chrome.i18n.getMessage(`feature_${feature}`) }
					</li>;
		});
	}

	render () {
		let { email, legal, year } = this.props.config;

		return <Text>
					<p className="tg-block">
						{ chrome.i18n.getMessage('description') }
					</p>

					<p className="tg-block">
						{ chrome.i18n.getMessage('features') }
					</p>

					<ul className="tg-list">
						{ this.getFeatures() }
					</ul>

					<p className="tg-block_small tg-center">
						{ `@ ${legal}, ${year}` }.
					</p>

					{/* <Share />*/}
				</Text>;
	}
}

About.propTypes = {
	config: PropTypes.object.isRequired
};

export default About;
