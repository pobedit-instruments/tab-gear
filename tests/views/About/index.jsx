import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';

import config from '../../../config';
import About from '../../../views/components/About';

test('About view', () => {
	let component = renderer.create(<About config={ config } />);
	let view = component.toJSON();

	let actual = expect(view);

	actual.toMatchSnapshot();
});
