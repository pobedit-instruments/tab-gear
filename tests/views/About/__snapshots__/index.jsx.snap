exports[`test About view 1`] = `
<div
  className="tg-box">
   
  <p
    className="tg-block">
    Tag Gears is an extension for Google Chrome that provides an intuitive tabs management.
  </p>
  <p
    className="tg-block">
    Some of the features that will come handy:
  </p>
  <ul
    className="tg-list">
    <li
      className="tg-list__item">
      List of all open tabs.
    </li>
    <li
      className="tg-list__item">
      Search by title and URL — just start typing.
    </li>
    <li
      className="tg-list__item">
      See incognito tabs.
    </li>
    <li
      className="tg-list__item">
      Tab discarding. Allows Chrome to automatically discard tabs that aren\'t of great interest to you when it\'s detected that system memory is running pretty low.
    </li>
    <li
      className="tg-list__item">
      Move your tabs in a preferred order — just drag and drop them!
    </li>
    <li
      className="tg-list__item">
      Closing selective tabs.
    </li>
    <li
      className="tg-list__item">
      Closing all open tabs.
    </li>
    <li
      className="tg-list__item">
      Displays the number of tabs you currently have open in all your Chrome windows.
    </li>
    <li
      className="tg-list__item">
      Quick opening your extensions.
    </li>
    <li
      className="tg-list__item">
      Localization for English and Russian languages.
    </li>
    <li
      className="tg-list__item">
      Press Ctrl+Shift+T or ⌘+⇧+T keys to show or hide the browser action popup.
    </li>
    <li
      className="tg-list__item">
      Do you want to prevent closing multiple tabs? It\'s possible by default and configurable!
    </li>
    <li
      className="tg-list__item">
      Highlighting your current and incognito tabs.
    </li>
  </ul>
  <p
    className="tg-block_small tg-center">
    @ Pobedit Instuments, 2016
    .
  </p>
   
</div>
`;
